sources/spanner/infoschema.go

```diff
@@ -349,19 +350,31 @@ func (isi InfoSchemaImpl) GetIndexes(conv *internal.Conv, table common.SchemaAnd
 		if err != nil {
 			return nil, fmt.Errorf("couldn't read row while fetching interleaved tables: %w", err)
 		}
-		err = row.Columns(&name, &column, &sequence, &ordering, &isUnique)
-		if err != nil {
-			fmt.Println(err)
-			conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
-			continue
+		if isi.SpDialect == constants.DIALECT_POSTGRESQL {
+			err = row.Columns(&name, &column, &sequence, &ordering, &isPgUnique)
+			if err != nil {
+				fmt.Println(err)
+				conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
+				continue
```

Comment:
This structure is not correct, we need to think of a better way to write this code where PGSQL specific nuances are not implemented as if statements - this is very error prone.

Please create a tech debt item with all the details.

Reply:
Created a ticket with id 315784385